e2e/consultemplate/consultemplate_test.go

```diff
@@ -0,0 +1,547 @@
+			found = true
+			break
+		}
+	}
+	must.True(t, found, must.Sprint("alloc failed but NOT due to expected source path escape error"))
+}
+
+// TestTemplatePathInterpolation_SharedAllocDir asserts that NOMAD_ALLOC_DIR
+// is supported as a destination for artifact and template blocks, and
+// that it is properly interpolated for task drivers with varying
+// filesystem isolation
+func TestTemplatePathInterpolation_SharedAllocDir(t *testing.T) {
+	cleanupGC(t)
+	submission, cleanupJob := jobs3.Submit(t,
+		"./input/template_shared_alloc.nomad",
+		jobs3.Timeout(time.Second*60)) // note: exec tasks can take a while
+	t.Cleanup(cleanupJob)
+	allocID := submission.AllocID("template-paths")
+
+	for _, task := range []string{"docker", "exec", "raw_exec"} {
+
+		// tests that we can render templates into the shared alloc directory
+		must.NoError(t, waitForTaskFile(allocID, task, "${NOMAD_ALLOC_DIR}/raw_exec.env",
+			func(out string) bool {
+				return len(out) > 0 && strings.TrimSpace(out) != "/alloc"
+			}, nil), must.Sprint("expected raw_exec.env to not be '/alloc'"))
+
+		must.NoError(t, waitForTaskFile(allocID, task, "${NOMAD_ALLOC_DIR}/exec.env",
+			func(out string) bool {
+				return strings.TrimSpace(out) == "/alloc"
+			}, nil), must.Sprint("expected shared exec.env to contain '/alloc'"))
+
+		must.NoError(t, waitForTaskFile(allocID, task, "${NOMAD_ALLOC_DIR}/docker.env",
+			func(out string) bool {
+				return strings.TrimSpace(out) == "/alloc"
+			}, nil), must.Sprint("expected shared docker.env to contain '/alloc'"))
+
+		// test that we can fetch artifacts into the shared alloc directory
+		for _, a := range []string{"google1.html", "google2.html", "google3.html"} {
+			must.NoError(t, waitForTaskFile(allocID, task, "${NOMAD_ALLOC_DIR}/"+a,
+				func(out string) bool {
+					return len(out) > 0
+				}, nil), must.Sprint("expected artifact in alloc dir"))
+		}
+
+		// test that we can load environment variables rendered with templates using interpolated paths
+		out, err := e2eutil.Command("nomad", "alloc", "exec", "-task", task, allocID, "sh", "-c", "env")
+		must.NoError(t, err)
+		must.StrContains(t, out, "HELLO_FROM=raw_exec")
+	}
+}
+
+// TestConsulTemplate_NomadServiceLookups tests consul-templates Nomad service
+// lookup functionality. It runs a job which registers two services, then
+// another which performs both a list and read template function lookup against
+// registered services.
+func TestConsulTemplate_NomadServiceLookups(t *testing.T) {
+
+	cleanupGC(t)
+
+	// The service job will be the source for template data
+	serviceJobSubmission, cleanupJob := jobs3.Submit(t, "./input/nomad_provider_service.nomad")
+	t.Cleanup(cleanupJob)
+	serviceAllocID := serviceJobSubmission.AllocID("nomad_provider_service")
+
+	// Create a non-default namespace.
+	_, err := e2eutil.Command("nomad", "namespace", "apply", "platform")
+	must.NoError(t, err)
+	cleanupNamespace(t, "platform")
```

Comment:
`namespace3` package has a thing for this which ain't bad

```go
t.Cleanup(namespaces3.Create(t, "platform"))
```

though it does use the `api` package rather than CLI, and I do think it's nice for e2e tests to use CLI...

for that, could use these new critters:

```go
e2eutil.MustCommand(t, "nomad namespace apply platform")
e2eutil.CleanupCommand(t, "nomad namespace delete platform")
```

but that's basically what you're doing here, so it's really 6 of one, half a dozen of the other.

Reply:
Used the `namespaces3` helper. Using the CLI is good for E2E tests but in this particular case there's no output from the command so we don't get much by exercising it.